prusti-viper/src/encoder/errors/error_manager.rs

```diff
@@ -217,18 +217,22 @@ impl<'tcx> ErrorManager<'tcx> {
     pub fn set_error(&mut self, pos: Position, error_ctxt: ErrorCtxt) {
         trace!("Register error {:?} at position id {:?}", error_ctxt, pos.id());
         assert_ne!(pos, Position::default(), "Trying to register an error on a default position");
-        if let Some(existing_error_ctxt) = self.error_contexts.get(&pos.id()) {
-            debug_assert_eq!(
-                existing_error_ctxt, &error_ctxt,
-                "An existing error context would be overwritten.\n\
-                Position id: {}\n\
-                Existing error context: {:?}\n\
-                New error context: {:?}",
-                pos.id(),
-                existing_error_ctxt,
-                error_ctxt
-            );
-        }
+        // TODO: disabled in #1284: when a predicate consists only of a call to
+        //   another function, the span of the call (PureFunctionCall) and the
+        //   span of the predicate definition (PureFunctionDefinition)
+        //   are the same
```

Comment:
Even if the span is the same, can't we encode them with different ids?

Reply:
Hmm, yes. For some reason when encoding predicates the existing `.pos()` was used instead of registering a new one, which also caused this problem. Now I am creating a new position. I seem to remember there was a reason the position was assigned this way but let's see CI…